ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -104,7 +104,9 @@ public static boolean isWindowBound(final ColumnName columnName) {
 
   @VisibleForTesting
   static boolean isPseudoColumn(final ColumnName columnName, final int pseudoColumnVersion) {
-    return pseudoColumnNames(pseudoColumnVersion).contains(columnName);
+    return pseudoColumns
+        .stream()
+        .anyMatch(col -> col.name.equals(columnName) && col.version <= pseudoColumnVersion);
```

Comment:
nit: we can make this slightly more readable by turning the second part of the boolean expression into a call to `filter()` before `anyMatch()`.